aspnetcore/security/anti-request-forgery.md

```diff
@@ -4,142 +4,230 @@ author: steve-smith
+Cookie-based authentication is a popular form of authentication. Token-based authentication systems are growing in popularity, especially for Single Page Applications (SPAs).
+
+### Cookie-based authentication
+
+When a user authenticates using their username and password, they're issued a token, containing an authentication ticket that can be used for authentication and authorization. The token is stored as a cookie that accompanies every request the client makes. Generating and validating this cookie is performed by the Cookie Authentication Middleware. The [middleware](xref:fundamentals/middleware/index) serializes a user principal into an encrypted cookie. On subsequent requests, the middleware validates the cookie, recreates the principal, and assigns the principal to the [User](/dotnet/api/microsoft.aspnetcore.http.httpcontext.user) property of [HttpContext](/dotnet/api/microsoft.aspnetcore.http.httpcontext).
 
-ASP.NET Core anti-request-forgery default data protection configuration 
+When a user is authenticated to access an app, a server-side user session is created and stored in a database or some other persistent store. The app generates a session key that points to the stored session in the data store and includes it in the authentication ticket/cookie. The web server checks this session key each time a user requests a resource that requires authorization. The app checks whether the associated user session has sufficient privileges to access the requested resource. If the user is authorized to access the resource, the request continues. Otherwise, the request returns as unauthorized. Cookies are used to make the app appear stateful, since the app is able to maintain the user's authentication/authorization state on each request.
+
+### Token-based authentication
+
+Token-based authentication doesn't store session on the server. When a user is authenticated, they're issued a token (not an antiforgery token). The token contains user information in the form of [claims](/dotnet/framework/security/claims-based-identity-model). When a user attempts to access a resource requiring authentication, the token is sent to the app with an additional authorization header in form of Bearer token. This makes the app stateless. In each subsequent request, the token is passed in the request for server-side validation. This token isn't *encrypted*; it's *encoded*. On the server, the token is decoded to access its information. To send the token on subsequent requests, either store the token in the browser's local storage or in a cookie. Don't be concerned about CSRF vulnerability if the token is stored in the browser's local storage. CSRF is a concern when the token is stored in a cookie.
```

Comment:
Again a potential mix up with "session". Also tokens do not have contain information about the user, it could be a reference token which points the application to some user state within itself (probably in a ... err ... session)

Reply:
Yes, this was all held over from the original topic. https://docs.microsoft.com/en-us/aspnet/core/security/anti-request-forgery#cookie-based-authentication

> tokens do not have contain information about the user, it could be a reference token which points the application to some user state within itself (probably in a ... err ... session)

I'll take a stab at updating it along those lines.